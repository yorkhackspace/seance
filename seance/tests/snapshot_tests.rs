#[test]
fn logo() {
    let design_file = seance::svg::parse_svg(
        &std::fs::read("../logo.svg").expect("failed to read logo SVG file"),
    )
    .expect("failed to parse logo SVG data");

    let design_name = "Logo";
    let offset = seance::DesignOffset { x: 0.0, y: 0.0 };

    let mut tool_passes = seance::default_passes::default_passes();
    for tool in &mut tool_passes {
        tool.set_enabled(true);
    }

    let paths = seance::svg::get_paths_grouped_by_colour(&design_file);
    let mut paths_in_mm = seance::resolve_paths(&paths, &offset, 1.0);
    seance::paths::filter_paths_to_tool_passes(&mut paths_in_mm, &tool_passes);
    let resolved_paths = seance::paths::convert_points_to_plotter_units(&paths_in_mm);
    insta::assert_debug_snapshot!("logo plotted paths", &resolved_paths);

    let hpgl = seance::hpgl::generate_hpgl(&resolved_paths, &tool_passes);
    insta::assert_snapshot!("logo HPGL output", &hpgl);

    let pcl = seance::pcl::wrap_hpgl_in_pcl(hpgl, design_name, &tool_passes);
    insta::assert_snapshot!("logo PCL output", &pcl);
}

#[test]
fn black_rectangle() {
    let design_file = seance::svg::parse_svg(
        &std::fs::read("tests/black_rectangle.svg").expect("failed to read rectangle SVG file"),
    )
    .expect("failed to parse rectangle SVG data");

    let design_name = "rectangle";
    let offset = seance::DesignOffset { x: 0.0, y: 0.0 };

    let mut tool_passes = seance::default_passes::default_passes();
    for tool in &mut tool_passes {
        tool.set_enabled(true);
    }

    let paths = seance::svg::get_paths_grouped_by_colour(&design_file);
    let mut paths_in_mm = seance::resolve_paths(&paths, &offset, 1.0);
    seance::paths::filter_paths_to_tool_passes(&mut paths_in_mm, &tool_passes);
    let resolved_paths = seance::paths::convert_points_to_plotter_units(&paths_in_mm);
    insta::assert_debug_snapshot!("rectangle plotted paths", &resolved_paths);

    let hpgl = seance::hpgl::generate_hpgl(&resolved_paths, &tool_passes);
    insta::assert_snapshot!("rectangle HPGL output", &hpgl);

    let pcl = seance::pcl::wrap_hpgl_in_pcl(hpgl, design_name, &tool_passes);
    insta::assert_snapshot!("rectangle PCL output", &pcl);
}
