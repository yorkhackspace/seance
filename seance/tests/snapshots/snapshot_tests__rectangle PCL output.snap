---
source: seance/tests/snapshot_tests.rs
expression: "&pcl"
---
%-12345XE!m9Nrectangle!v16R1111111111111111!v64I0400040004000400040004000400040004000400040004000400040004000400!v64V0020002000200020002000200020002000200020002000200020002000200020!v64P0100010001000100010001000100010001000100010001000100010001000100!v16D*t508R&u508R!r0N%1A!r1000I!r1000K!r500P*t508R&u508R!m0S!s1S%1BIN;SC;PU;SP1;LT;PU0,18528;SP1;PU3004,15124;PD3004,15124;PD3044,15124;PD3084,15124;PD3124,15124;PD3164,15124;PD3204,15124;PD3244,15124;PD3284,15124;PD3324,15124;PD3364,15124;PD3404,15124;PD3444,15124;PD3484,15124;PD3524,15124;PD3564,15124;PD3604,15124;PD3644,15124;PD3684,15124;PD3724,15124;PD3764,15124;PD3804,15124;PD3844,15124;PD3884,15124;PD3924,15124;PD3964,15124;PD4004,15124;PD4044,15124;PD4084,15124;PD4124,15124;PD4164,15124;PD4204,15124;PD4244,15124;PD4284,15124;PD4324,15124;PD4364,15124;PD4404,15124;PD4444,15124;PD4484,15124;PD4524,15124;PD4564,15124;PD4604,15124;PD4644,15124;PD4684,15124;PD4724,15124;PD4764,15124;PD4804,15124;PD4844,15124;PD4884,15124;PD4924,15124;PD4964,15124;PD5004,15124;PD5044,15124;PD5084,15124;PD5124,15124;PD5164,15124;PD5204,15124;PD5244,15124;PD5284,15124;PD5324,15124;PD5364,15124;PD5404,15124;PD5444,15124;PD5484,15124;PD5524,15124;PD5564,15124;PD5604,15124;PD5644,15124;PD5684,15124;PD5724,15124;PD5764,15124;PD5804,15124;PD5844,15124;PD5871,15111;PD5871,15071;PD5871,15031;PD5871,14991;PD5871,14951;PD5871,14911;PD5871,14871;PD5871,14831;PD5871,14791;PD5871,14751;PD5871,14711;PD5871,14671;PD5871,14631;PD5871,14591;PD5871,14551;PD5871,14511;PD5871,14471;PD5871,14431;PD5871,14391;PD5871,14351;PD5871,14311;PD5871,14271;PD5871,14231;PD5871,14191;PD5871,14151;PD5871,14111;PD5871,14071;PD5871,14031;PD5871,13991;PD5871,13951;PD5871,13911;PD5871,13871;PD5871,13831;PD5871,13791;PD5871,13751;PD5871,13711;PD5871,13671;PD5871,13631;PD5871,13591;PD5871,13551;PD5871,13511;PD5871,13471;PD5871,13431;PD5871,13391;PD5871,13351;PD5871,13311;PD5871,13271;PD5871,13231;PD5871,13191;PD5871,13151;PD5871,13111;PD5871,13071;PD5871,13031;PD5871,12991;PD5871,12951;PD5871,12911;PD5871,12871;PD5871,12831;PD5871,12791;PD5871,12751;PD5871,12711;PD5871,12671;PD5871,12631;PD5871,12591;PD5871,12551;PD5871,12511;PD5871,12471;PD5871,12431;PD5871,12391;PD5871,12351;PD5871,12311;PD5871,12271;PD5871,12231;PD5871,12191;PD5871,12151;PD5839,12143;PD5799,12143;PD5759,12143;PD5719,12143;PD5679,12143;PD5639,12143;PD5599,12143;PD5559,12143;PD5519,12143;PD5479,12143;PD5439,12143;PD5399,12143;PD5359,12143;PD5319,12143;PD5279,12143;PD5239,12143;PD5199,12143;PD5159,12143;PD5119,12143;PD5079,12143;PD5039,12143;PD4999,12143;PD4959,12143;PD4919,12143;PD4879,12143;PD4839,12143;PD4799,12143;PD4759,12143;PD4719,12143;PD4679,12143;PD4639,12143;PD4599,12143;PD4559,12143;PD4519,12143;PD4479,12143;PD4439,12143;PD4399,12143;PD4359,12143;PD4319,12143;PD4279,12143;PD4239,12143;PD4199,12143;PD4159,12143;PD4119,12143;PD4079,12143;PD4039,12143;PD3999,12143;PD3959,12143;PD3919,12143;PD3879,12143;PD3839,12143;PD3799,12143;PD3759,12143;PD3719,12143;PD3679,12143;PD3639,12143;PD3599,12143;PD3559,12143;PD3519,12143;PD3479,12143;PD3439,12143;PD3399,12143;PD3359,12143;PD3319,12143;PD3279,12143;PD3239,12143;PD3199,12143;PD3159,12143;PD3119,12143;PD3079,12143;PD3039,12143;PD3004,12147;PD3004,12187;PD3004,12227;PD3004,12267;PD3004,12307;PD3004,12347;PD3004,12387;PD3004,12427;PD3004,12467;PD3004,12507;PD3004,12547;PD3004,12587;PD3004,12627;PD3004,12667;PD3004,12707;PD3004,12747;PD3004,12787;PD3004,12827;PD3004,12867;PD3004,12907;PD3004,12947;PD3004,12987;PD3004,13027;PD3004,13067;PD3004,13107;PD3004,13147;PD3004,13187;PD3004,13227;PD3004,13267;PD3004,13307;PD3004,13347;PD3004,13387;PD3004,13427;PD3004,13467;PD3004,13507;PD3004,13547;PD3004,13587;PD3004,13627;PD3004,13667;PD3004,13707;PD3004,13747;PD3004,13787;PD3004,13827;PD3004,13867;PD3004,13907;PD3004,13947;PD3004,13987;PD3004,14027;PD3004,14067;PD3004,14107;PD3004,14147;PD3004,14187;PD3004,14227;PD3004,14267;PD3004,14307;PD3004,14347;PD3004,14387;PD3004,14427;PD3004,14467;PD3004,14507;PD3004,14547;PD3004,14587;PD3004,14627;PD3004,14667;PD3004,14707;PD3004,14747;PD3004,14787;PD3004,14827;PD3004,14867;PD3004,14907;PD3004,14947;PD3004,14987;PD3004,15027;PD3004,15067;PD3004,15107;PD3004,15124;PU0,18528;SP0;EC0;EC1;OE;%1AE%-12345X
